### PR TITLE
Ensure proper permissions when accessing the "Refresh Opportunity Names" page in NPSP Settings

### PR DESCRIPTION
**Pull request by force2b**
# Critical Changes

- In order to use the "Refresh Opportunity Names" page in NPSP Settings, the User must have Update access to the Opportunity object as well as Read access to both the Account and Contact objects.

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by force2b** [comment]
[W-8428988](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008iqpvIAA/view)

**Review by andrewyu-salesforce — APPROVED** [review]
@force2b everything looks great.

**Review by npsp-reedestockton — CHANGES_REQUESTED** [review]
Looks great!  Just a couple of small formatting comments.

File: src/classes/STG_PanelOppNamingBatch_CTRL.cls [hunk 1]
```diff
@@ -36,30 +36,57 @@
...
         isRunningBatch = false;
     }
 
     /*********************************************************************************************************
-    * @description Returns the string Id of the Households panel. 
+    * @description NPSP namespace: an empty string if unmanaged, or 'npsp' if managed
+    */
+    public String getNamespace() {
+        String namespace = UTIL_Namespace.getNamespace();
+        if (String.isBlank(namespace)) return 'c';
```
[the remark this answers is not in the document]
Comment (force2b) [reply src/classes/STG_PanelOppNamingBatch_CTRL.cls:50]: Definitely. Good catch. I'll fix the class I copied it from as well.

**Review by npsp-reedestockton — APPROVED** [review]
Nice!  Thanks!

**Pull request merged**